### PR TITLE
Add MassConcentration, MolarConcentration, and CatalyticActivityConcentration.

### PR DESCRIPTION
**Pull request by okkn**
Hi. I added MassConcentration (ex. kg/m³, mg/dL), MolarConcentration (mol/m³, mol/L), and CatalyticActivityConcentration (kat/m³, U/L) to this fabulous library. To separate MassConcentration from MassDensity, I introduced a new Kind trait (ConstituentConcentrationKind).
These quantities are very often used in medicine and biology.
Regards,

### CODE REVIEW

**Comment by iliekturtles** [comment]
Thanks for the PR! I'm going to try to review tomorrow.

File: src/si/catalytic_activity_concentration.rs [hunk 1]
```diff
@@ -0,0 +1,412 @@
...
+        @nanokatal_per_milliliter:
+            prefix!(nano) / prefix!(milli) / prefix!(milli);
+            "nkat/mL", "nanokatal per milliliter", "nanokatals per milliliter";
+        @picokatal_per_milliliter:
+            prefix!(pico) / prefix!(milli) / prefix!(milli);
+            "pkat/mL", "picokatal per milliliter", "picokatals per milliliter";
+        @femtokatal_per_milliliter:
+            prefix!(femto) / prefix!(milli) / prefix!(milli);
+            "fkat/mL", "femtokatal per milliliter", "femtokatals per milliliter";
+
```
Comment (iliekturtles) [inline src/si/catalytic_activity_concentration.rs]: Extra newline.

File: src/si/catalytic_activity_concentration.rs [hunk 2]
```diff
@@ -0,0 +1,412 @@
...
+            test::<ca::femtokatal, v::deciliter, c::femtokatal_per_deciliter>();
+
+            test::<ca::kilokatal, v::milliliter, c::kilokatal_per_milliliter>();
+            test::<ca::katal, v::milliliter, c::katal_per_milliliter>();
+            test::<ca::millikatal, v::milliliter, c::millikatal_per_milliliter>();
+            test::<ca::microkatal, v::milliliter, c::microkatal_per_milliliter>();
+            test::<ca::nanokatal, v::milliliter, c::nanokatal_per_milliliter>();
+            test::<ca::picokatal, v::milliliter, c::picokatal_per_milliliter>();
+            test::<ca::femtokatal, v::milliliter, c::femtokatal_per_milliliter>();
+
```
Comment (iliekturtles) [inline src/si/catalytic_activity_concentration.rs]: Extra newline.

File: src/si/mass_concentration.rs [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+            "picogram per cubic meter", "picograms per cubic meter";
+        @femtogram_per_cubic_meter: prefix!(femto) / prefix!(kilo); "fg/m³",
+            "femtogram per cubic meter", "femtograms per cubic meter";
+        @attogram_per_cubic_meter: prefix!(atto) / prefix!(kilo); "ag/m³",
+            "attogram per cubic meter", "attograms per cubic meter";
+        @zeptogram_per_cubic_meter: prefix!(zepto) / prefix!(kilo); "zg/m³",
+            "zeptogram per cubic meter", "zeptograms per cubic meter";
+        @yoctogram_per_cubic_meter: prefix!(yocto) / prefix!(kilo); "yg/m³",
+            "yoctogram per cubic meter", "yoctograms per cubic meter";
+
```
Comment (iliekturtles) [inline src/si/mass_concentration.rs]: Extra newline.

File: src/si/mass_concentration.rs [hunk 4]
```diff
@@ -0,0 +1,228 @@
...
+        @nanogram_per_milliliter:
+            (prefix!(nano) / prefix!(kilo)) / prefix!(milli) / prefix!(milli);
+            "ng/mL", "nanogram per milliliter", "nanograms per milliliter";
+        @picogram_per_milliliter:
+            (prefix!(pico) / prefix!(kilo)) / prefix!(milli) / prefix!(milli);
+            "pg/mL", "picogram per milliliter", "picograms per milliliter";
+        @femtogram_per_milliliter:
+            (prefix!(femto) / prefix!(kilo)) / prefix!(milli) / prefix!(milli);
+            "fg/mL", "femtogram per milliliter", "femtograms per milliliter";
+
```
Comment (iliekturtles) [inline src/si/mass_concentration.rs]: Extra newline.

File: src/si/molar_concentration.rs [hunk 5]
```diff
@@ -0,0 +1,227 @@
...
+        @nanomole_per_milliliter:
+            prefix!(nano) / prefix!(milli) / prefix!(milli);
+            "nmol/mL", "nanomole per milliliter", "nanomoles per milliliter";
+        @picomole_per_milliliter:
+            prefix!(pico) / prefix!(milli) / prefix!(milli);
+            "pmol/mL", "picomole per milliliter", "picomoles per milliliter";
+        @femtomole_per_milliliter:
+            prefix!(femto) / prefix!(milli) / prefix!(milli);
+            "fmol/mL", "femtomole per milliliter", "femtomoles per milliliter";
+
```
Comment (iliekturtles) [inline src/si/molar_concentration.rs]: Extra newline.

**Comment by okkn** [comment]
As you indicated, these newlines are extra. I removed them.

**Comment by iliekturtles** [comment]
One final line to remove shown in the diff below (ignore the encoding issues). If you could squash the fixes into the original commits I'll merge. Will likely look at the build timeout as a separate issue.

```diff
diff --git a/src/si/catalytic_activity_concentration.rs b/src/si/catalytic_activity_concentration.rs
index 638bb94..e44a1a3 100644
--- a/src/si/catalytic_activity_concentration.rs
+++ b/src/si/catalytic_activity_concentration.rs
@@ -4,7 +4,6 @@ quantity! {
     /// Catalytic activity concentration (base unit katal per cubic meter, mol ┬╖ sΓü╗┬╣ ┬╖ mΓü╗┬│).
     quantity: CatalyticActivityConcentration; "catalytic activity concentration";
     /// Dimension of catalytic activity concentration, LΓü╗┬│TΓü╗┬╣N (base unit katal, mol ┬╖ sΓü╗┬╣ ┬╖ mΓü╗┬│).
-    /// (base unit katal per cubic meter, mol ┬╖ sΓü╗┬╣ ┬╖ mΓü╗┬│).
     dimension: ISQ<
         N3,     // length
         Z0,     // mass
```

**Comment by okkn** [comment]
Fixed😉

**Comment by iliekturtles** [comment]
I squashed the fixes into the original commits and merged. Thanks so much for the PR!

**Pull request closed without merging**